content/blog/readr-2-0-0/index.Rmd

```diff
@@ -0,0 +1,265 @@
+immediately, only when they are actually read.
+Because of this a warning will be issued the first time a problem is encountered,
+which may happen after initial reading.
+
+Run `problems()` on your dataset to read the entire dataset and return all of the problems found.
+Run `problems(lazy = TRUE)` if you only want to retrieve the problems found so far.
+
+Deleting files after reading is also impacted by laziness.
+On Windows open files cannot be deleted as long as a process has the file open.
+Because readr keeps a file open when reading lazily this means you cannot read, then immediately delete the file.
+readr will in most cases close the file once it has been completely read.
+However, if you know you want to be able to delete the file after reading it is best to pass `lazy = FALSE` when reading the file.
+
+## Control over quoting and escaping when writing
+
+You can now explicitly control how fields are quoted and escaped when writing with the `quote` and `escape` arguments to `write_*()` functions.
+
+`quote` has three options.
+
+1. 'needed' - which will quote fields only when needed.
+2. 'all' - which will always quote all fields.
+3. 'none' - which will never quote any fields.
+
+`escape` also has three options, to control how quote characters are escaped.
+
+1. 'double' - which will use double quotes to escape quotes.
+2. 'backslash' - which will use a backslash to escape quotes.
+3. 'none' - which will not do anything to escape quotes.
+
+We hope these options will give people the flexibility they need when writing files using readr.
+
+## Literal data
+
+In edition one the reading functions treated any input with a newline in it or vectors of length > 1 as literal data.
+In edition two vectors of length > 1 are nowassumed to correspond to multiple files.
+Because of this we now have a more explicit way to represent literal data, by putting `I()` around the input.
+
+```{r}
+readr::read_csv(I("a,b\n1,2"))
+```
+
+## Lighter installation requirements
+
+readr now should be much easier to install.
+Previous versions of readr used the Boost C++ library to do some of the numeric parsing.
+While these are well written, robust libraries, the BH package which contains them has a large number of files (1500+) which can take a long time to install.
+In addition the code within these headers is complicated and can take a large amount of memory (2+ Gb) to compile, which made it challenging to compile readr from source in some cases.
+
+readr no longer depends on Boost or the BH package, so should compile more quickly in most cases.
+
+## Deprecated functions and features
+
+* `melt_csv()`, `melt_delim()`, `melt_tsv()` and `melt_fwf()` have been deprecated.
+  These functions rely on the first edition parsing code and would be challenging to update to the new parser.
+  When the first edition parsing code is eventually removed from readr they will be split off into a new package.
```

Comment:
Does this mean the functions are superseded rather than deprecated? My main assumption about deprecated functions is that they will soon no longer work and I need to find an alternative. However we seem to be committed to maintaining a superseded package here.

Reply:
Yeah, this needs a little tweak of the language.